CRM/Contact/BAO/Relationship.php

```diff
@@ -1747,9 +1708,9 @@ public static function relatedMemberships($contactId, &$params, $ids, $action =
 
             if ($action & CRM_Core_Action::UPDATE) {
```

Comment:
The action is `CRM_Core_Action::ADD` when adding a new relationship - that makes the different behavior between adding and re-enabling

Reply:
Hmm - looks like we just need to make that UPDATE OR ADD - I hate the bitwise operator